ee/surveys/summaries/summarize_surveys.py

```diff
@@ -0,0 +1,131 @@
+import json
+
+import openai
+
+from datetime import datetime
+from typing import Optional
+
+from posthog.hogql import ast
+from posthog.hogql.parser import parse_select
+from posthog.hogql_queries.insights.paginators import HogQLHasMorePaginator
+from posthog.schema import HogQLQueryResponse
+from posthog.utils import get_instance_region
+
+from prometheus_client import Histogram
+
+from posthog.api.activity_log import ServerTimingsGathered
+from posthog.models import Team, User
+
+import structlog
+
+logger = structlog.get_logger(__name__)
+
+TOKENS_IN_PROMPT_HISTOGRAM = Histogram(
+    "posthog_survey_summary_tokens_in_prompt_histogram",
+    "histogram of the number of tokens in the prompt used to generate a survey summary",
+    buckets=[
+        0,
+        10,
+        50,
+        100,
+        500,
+        1000,
+        2000,
+        3000,
+        4000,
+        5000,
+        6000,
+        7000,
+        8000,
+        10000,
+        20000,
+        30000,
+        40000,
+        50000,
+        100000,
+        128000,
+        float("inf"),
+    ],
+)
+
+
+def prepare_data(query_response: HogQLQueryResponse) -> list[str]:
+    response_values = []
+    properties_list: list[dict] = [json.loads(x[1]) for x in query_response.results]
+    for props in properties_list:
+        response_values.extend([value for key, value in props.items() if key.startswith("$survey_response") and value])
+    return response_values
+
+
+def summarize_survey_responses(
+    survey_id: str, question_index: Optional[int], survey_start: datetime, survey_end: datetime, team: Team, user: User
+):
+    timer = ServerTimingsGathered()
+
+    with timer("prepare_query"):
+        paginator = HogQLHasMorePaginator(limit=100, offset=0)
+        q = parse_select(
+            """
+            SELECT distinct_id, properties, person.properties
```

Comment:
this is straight copy from the FE, it doesn't need to load all properties like this
(maybe it doesn't in the FE either 🤔)